Sources/Disposable.swift

```diff
@@ -350,10 +350,10 @@ public final class SerialDisposable: Disposable {
 		}
 
 		set(disposable) {
-			_inner.swap(disposable)?.dispose()
-
-			if let disposable = disposable, isDisposed {
-				disposable.dispose()
+			if isDisposed {
+				disposable?.dispose()
+			} else {
+				_inner.swap(disposable)?.dispose()
```

Comment:
We need to check `isDisposed` again after swapping, since `self` could be disposed of in parallel during the process, racing against https://github.com/ReactiveCocoa/ReactiveSwift/pull/688/files#diff-7056bb5d45375e26f2eeb3adeabac16bR373.

Reply:
Change suggestions do not seem to be available in this repo.

Essentially, we need to do this after the swapping in the else path:

```
if isDisposed {
    _inner.swap(nil)?.dispose()
}
```

Because `dispose()` can be called in parallel to `inner = new` on different threads. Depending on the order observed by the system memory, it could yield a boundary case where `isDisposed` is `true` but the inner disposable is left untouched. The conflicting path may 

> I always thought that such operations should not be done in a multi-threading fashion, so it will be the users responsibility to serialize correctly.

Disposables have always meant to be thread safe, despite us not explicitly having written this as part of the contract.

Reply:
> Change suggestions do not seem to be available in this repo

Weird. I don't see a setting for it.